app/api/v3/endpoints/ad_element_extractor_router.py

```
@@ -0,0 +1,12 @@
+from fastapi import APIRouter, HTTPException
+from app.schemas.v2.ad_element_extractor_dto import AdElementDTOV2
+from app.services.v2.ad_element_extractor_service import AdElementExtractorServiceV2
+
+router = APIRouter()
+
+@router.post("/extract", response_model=AdElementDTOV2.AdElementResponse)
+async def extract_ad_elements(req: AdElementDTOV2.AdElementRequest):
+    try:
+        return AdElementExtractorServiceV2.extract_elements(req)
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)